src/ledger/events.rs

```diff
@@ -0,0 +1,109 @@
+use chrono::DateTime;
+use chrono::Utc;
+use ethereum_types::U256;
+use revm::primitives::Address;
+use uuid::Uuid;
+
+use crate::eth::primitives::BlockNumber;
+use crate::eth::primitives::Hash;
+
+/// Represents token transfers (debits and credits) associated with a specific Ethereum account within a single transaction.
+///
+/// The `primary_account_address` field identifies the primary account involved in these transfers.
+///
+/// An event will be generated for each account involved in a transaction, meaning if a transaction is not the primary in this event,
+/// in another event it will treated as the primary and credit and debit operations adjusted accordingly.
```

Comment:
Stratus will use only the `Transfer` events emitted by ERC-20 contracts.

It will generate one event payload per (transaction hash + account) and not one payload event per transaction, so batch transactions does not really changes anything.

Reply:
So for a batch transaction with size 5, we'd emit 6 events:

- 1 with where the primaryAccount is the debitParty.
- 5 where the primaryAccount is the creditParty.

Got it, thanks for clarifying!